compiler/rustc_hir_typeck/src/opaque_types.rs

```diff
@@ -88,19 +112,47 @@ impl<'tcx> FnCtxt<'_, 'tcx> {
                 _ => unreachable!("not opaque or generator: {def_id:?}"),
             }
 
+            // We do actually need to check this the second pass (we can't just
+            // store this), because we can go from `UnconstrainedHiddenType` to
+            // `HasDefiningUse` (because of fallback)
```

Comment:
we can also just go from no uses to a defining use due to fallback, e.g. if you have soem `?will_fallback: Trait<opaque>` goal